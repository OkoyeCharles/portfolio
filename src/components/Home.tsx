import React from "react";
import OkoyeCharles from "../images/personal/OkoyeCharles.webp";
import OkoyeCharlesSm from "../images/personal/OkoyeCharles-sm.webp";
import { Reveal } from "react-awesome-reveal";
import { keyframes } from "@emotion/react";

const slideIn = keyframes`
  from {
    opacity: 0;
    transform: translateX(3em);
  }

  to {
    opacity: 1;
    transform: translateX(0);
  }
`;

const slideInL = keyframes`
  from {transform: translateX(-5em)}
  to {transform: translateX(0)}
`;

interface HomeProps {
  mediaWidth: number;
}

const Home: React.FC<HomeProps> = ({ mediaWidth }) => {
  return (
    <section className="home">
      <div className="home__container">
        {mediaWidth > 800 ? (
          <div className="home__content">
            <Reveal triggerOnce keyframes={slideIn} cascade damping={0.25}>
              <h1 className="home__header">OKOYE CHARLES</h1>
              <div className="home__description">
                My Name is Okoye Charles and I'm a software developer! I can
                help you build a product , feature or website Look through some
                of my work and experience! If you like what you see and need
                help on a project, I am available for hire.
              </div>
              <a href="#projects">
                <button type="button" className="home__button">
                  MY PROJECTS
                </button>
              </a>
            </Reveal>
          </div>
        ) : (
          <div className="home__content">
            <h1 className="home__header">OKOYE CHARLES</h1>
            <div className="home__description">
              My Name is Okoye Charles and I'm a software developer! I can help
              you build a product , feature or website Look through some of my
              work and experience! If you like what you see and need help on a
              project, I am available for hire.
            </div>
            <a href="#projects">
              <button type="button" className="home__button">
                MY PROJECTS
              </button>
            </a>
          </div>
        )}

        {mediaWidth > 800 ? (
          <div className="home__image">
            <Reveal triggerOnce keyframes={slideInL} damping={0.25}>
              <img
                src={OkoyeCharles}
                width="2960"
                height="2960"
                alt="Okoye Charles"
              />
            </Reveal>
          </div>
        ) : (
          <div className="home__image">
            <img src={OkoyeCharlesSm} 
            width="800"
            height="800"
            alt="Okoye Charles" />
          </div>
        )}
      </div>
    </section>
  );
};

export default Home;
