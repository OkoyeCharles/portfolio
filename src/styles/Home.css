.home {
  min-height: 90vh;
  display: flex;
  align-items: flex-end;
  max-width: 1600px;
  margin-inline: auto;
}

.home__container {
  padding: 0 1.8em 0 4.5em;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: -2em;
}

/* .home, .home * {
  box-shadow: 0 0 0 .1em red;
} */

.home__content {
  grid-column: 1/span 3;
  grid-row: 1;
}

.home__header {
  font-size: clamp(3em, 7.25vw, 6.25em);
  font-weight: 900;
  color: #111;
  margin-top: min(2vw, 2rem);
}

.home__description {
  width: 75%;
  margin-top: .5em;
  font-size: 1.25em;
  font-weight: 300;
}

.home__button {
  position: relative;
  margin-block: 1.75em;
  border: 0.075em solid #111;
  padding: 0.5em 2em;
  cursor: pointer;
  font-weight: 300;
  overflow: hidden;
  background: #f0f0f0;
  transition: .1s ease;
}

.home__button:is(:hover, :focus-within) {
  background: #222;
  color: rgba(255, 255, 255, 0.789);
  box-shadow: 0 0 .5em rgba(0, 0, 0, 0.3);
}

.home__image {
  position: relative;
  grid-column: 3/span 3;
  grid-row: 1;
  display: flex;
  justify-content: flex-end;
  align-items: flex-end;
  user-select: none;
}

.home__image img {
  width: 95%;
  margin-left: auto;
}

.home__image::after {
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

@media screen and (max-width: 800px) {
  .home {
    min-height: 0;
    padding-top: 1.5em;
    z-index: -1;
  }

  .home__container {
    display: flex;
    flex-direction: column;
    place-items: center;
  }

  .home__content {
    display: flex;
    flex-direction: column;
  }

  .home__header {
    font-size: clamp(1.5em,7vw, 2.5em);
  }
  /* .home__content > * {
  box-shadow: 0 0 0 .1em red;
  } */

  .home__description {
    width: 100%;
    font-size: clamp(1em, 3vw, 1.25em);
  }

  .home__content a {
    width: fit-content;
    margin-block: 1.75em;
    align-self: center;
  }
  
  .home__button {
    margin: 0;
    justify-self: center;
    z-index: 0;
  }

  .home__image {
    justify-content: center;
    
  }
  .home__image img {
    margin-inline: auto;
    width: 50%;
    min-width: 12.5em;
  }
}

@media screen and (max-width: 700px) {
  .home__container {
    padding: 0 1.8em 0 1.8em;
  }

  .home__button {
    z-index: -1;
  }

  .home__description {
    text-align: center;
  }

  .home__image {
    z-index: -1;
  }
}