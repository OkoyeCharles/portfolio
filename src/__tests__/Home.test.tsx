import renderer from 'react-test-renderer';
import Home from '../components/Home';

it('<Home /> component renders properly', () => {
  const component = renderer.create(
    <Home mediaWidth={200}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});