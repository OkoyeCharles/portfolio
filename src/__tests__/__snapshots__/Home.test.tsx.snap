// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> component renders properly 1`] = `
<section
  className="home"
>
  <div
    className="home__container"
  >
    <div
      className="home__content"
    >
      <h1
        className="home__header"
      >
        OKOYE CHARLES
      </h1>
      <div
        className="home__description"
      >
        My Name is Okoye Charles and I'm a software developer! I can help you build a product , feature or website Look through some of my work and experience! If you like what you see and need help on a project, I am available for hire.
      </div>
      <a
        href="#CONTACT"
      >
        <button
          className="home__button"
          type="button"
        >
          CONTACT ME
        </button>
      </a>
    </div>
    <div
      className="home__image"
    >
      <img
        alt="Okoye Charles"
        height="800"
        src="OkoyeCharles-sm.webp"
        width="800"
      />
    </div>
  </div>
</section>
`;
